data/vibe/data/json.d

```diff
@@ -1925,6 +1926,17 @@ struct JsonStringSerializer(R, bool pretty = false)
 	this(R range)
 	{
 		m_range = range;
+		static if (isOutputRange!(R, char))
+		{
+			dg = (scope const(char)[] data)
+			{
+				static if (__traits(compiles, m_range ~= data))
+					m_range ~= data;
+				else
+					foreach (const c; data)
+						m_range.put(c);
+			};
+		}
```

Comment:
Having this delegate in the ctor has the downside of creating an extra GC allocation and an extra reference to `m_range` through the context.
It would be much better to move this delegate to the function that pass it (`serializeSinkType`) and use `scope`.
But even better, you could just make it a private method and pass `&this.sink` at L2028 instead of `dg`.